HippolyteTests/JSONMatcherTests.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  JSONMatcherTests.swift
+//  HippolyteTests_iOS
+//
+//  Created by Clemens on 04.07.19.
+//  Copyright © 2019 Clemens Schulz. All rights reserved.
+//
+
+import XCTest
+import Hippolyte
+
+class JSONMatcherTests: XCTestCase {
+
+  private struct TestObject: Codable, Hashable {
+
+    var id: Int
+    var name: String?
+    var foo: Bool
+  }
+
+  func testMatchingObjectMatches() {
+    let object = TestObject(id: 1, name: "name", foo: false)
+    let matcher = JsonMatcher<TestObject>(object: object)
+
+    let encoder = JSONEncoder()
+    let data = try! encoder.encode(object)
+
+    XCTAssertTrue(matcher.matches(data: data))
+  }
+
+  func testMismatchingObjectDoesNotMatch() {
+    let object = TestObject(id: 1, name: "name", foo: false)
+    let matcher = JsonMatcher<TestObject>(object: object)
+
+    let otherObject = TestObject(id: 1, name: "other", foo: true)
+
+    let encoder = JSONEncoder()
+    let data = try! encoder.encode(otherObject)
```

Comment:
instead of using `try!` you can write your test as `func testMismatchingObjectDoesNotMatch() throws {` and then just use a regular `try` in here. The test would also nicely fail if it failed to encode